lib/charms/tempo_k8s/v1/charm_tracing.py

```diff
@@ -337,7 +457,18 @@ def wrap_init(self: CharmBase, framework: Framework, *args, **kwargs):
         # on the assumption that spans will be used as contextmanagers.
         # Since we don't (as we need to close the span on framework.commit),
         # we need to manually set the root span as current.
-        span = _tracer.start_span("charm exec", attributes={"juju.dispatch_path": dispatch_path})
+        attributes = {"juju.dispatch_path": dispatch_path}
+        if SNAPSHOT_CONFIG.active:
+            logger.debug("gathering state...")
+            state = _get_state(self)
+            if state:
+                attributes["state"] = state
+            else:
+                logger.debug(f"could not collect state for {self}")
+
+        logger.debug("initializing root 'charm exec' span")
```

Comment:
Not certain if that log adds context. It might if we debug-logged the created root span ID - but then we'd need to add the log after starting the span.